generator/src/main/java/com/scottlogic/deg/generator/walker/reductive/FieldCollectionHelper.java

```diff
@@ -0,0 +1,169 @@
+        FieldSpecMerger fieldSpecMerger,
+        FieldSpecFactory fieldSpecFactory,
+        FixFieldStrategy fixFieldStrategy,
+        ReductiveDataGeneratorMonitor monitor) {
+        this.fieldSpecMerger = fieldSpecMerger;
+        this.fieldSpecFactory = fieldSpecFactory;
+        this.fixFieldStrategy = fixFieldStrategy;
+        this.generationConfig = config;
+        this.reducer = constraintReducer;
+        this.monitor = monitor;
+    }
+
+
+    //produce a stream of RowSpecs for each value in the permitted set of values for the field fixed on the last iteration
+    public Stream<RowSpec> createRowSpecFromFixedValues(FieldCollection fieldCollection, ConstraintNode constraintNode) {
+        //create a row spec where every field is set to this.fixedFields & field=value
+        if (fieldCollection.getLastFixedField() == null) {
+            throw new UnsupportedOperationException("Field has not been fixed yet");
+        }
+
+        Map<Field, FieldSpec> fieldSpecsPerField = getFieldSpecsForAllFixedFieldsExceptLast(fieldCollection, constraintNode);
+
+        if (fieldSpecsPerField.values().stream().anyMatch(fieldSpec -> fieldSpec == FieldSpec.Empty)){
+            return Stream.empty();
+        }
+
+        FieldSpec fieldSpecForValuesInLastFixedField = fieldCollection.getLastFixedField().getFieldSpecForValues();
+        fieldSpecsPerField.put(fieldCollection.getLastFixedField().field, fieldSpecForValuesInLastFixedField);
+
+        RowSpec rowSpecWithAllValuesForLastFixedField = new ReductiveRowSpec(
+            fieldCollection.getFields(),
+            fieldSpecsPerField,
+            fieldCollection.getLastFixedField().field
+        );
+
+        this.monitor.rowSpecEmitted(
+            fieldCollection.getLastFixedField(),
+            fieldSpecForValuesInLastFixedField,
+            rowSpecWithAllValuesForLastFixedField);
+        return Stream.of(rowSpecWithAllValuesForLastFixedField);
+    }
+
+    //work out the next field to fix and return a new FieldCollection with this field fixed
+    public FieldCollection getNextFixedField(FieldCollection fieldCollection, ReductiveConstraintNode rootNode) {
+        Field fieldToFix = this.fixFieldStrategy.getNextFieldToFix(fieldCollection, rootNode);
+
+        if (fieldToFix == null){
+            throw new UnsupportedOperationException(
+                String.format(
+                    "Unable to find a field to fix, no finite constraints\nUnfixed fields: %s",
+                    Objects.toString(fieldCollection.getUnfixedFields())));
+        }
+
+        FixedField field = getFixedFieldWithValuesForField(fieldToFix, rootNode);
+        return fieldCollection.with(field);
+    }
+
+
+    //Given the current set of fixed fields, work out if the given atomic constraint is contradictory, whether the field is fixed or not
+    AtomicConstraintFixedFieldBehaviour shouldIncludeAtomicConstraint(FieldCollection fieldCollection, AtomicConstraint atomicConstraint) {
```

Comment:
This method could move to the `ReductiveDecisionTreeAdapter` since the previously private methods are now public, same as `fixedValueConflictsWithAtomicConstraint`